addons/provideContext.js

```diff
@@ -46,4 +39,39 @@ module.exports = function provideContext(Component, customContextTypes) {
     });
 
     return ContextProvider;
+}
+
+/**
+ * Provides context prop to all children as React context
+ *
+ * Example:
+ *   var WrappedComponent = provideContext(Component, {
+ *       foo: React.PropTypes.string
+ *   });
+ *
+ * Also supports the decorator pattern:
+ *   @provideContext({
+ *       foo: React.PropTypes.string
+ *   })
+ *   class ConnectedComponent extends React.Component {
+ *       render() {
+ *           return <div/>;
+ *       }
+ *   }
+ *
+ * @method provideContext
+ * @param {React.Component} [Component] component to wrap
+ * @param {object} customContextTypes Custom contextTypes to add
+ * @returns {React.Component} or {Function} if using decorator pattern
+ */
+module.exports = function provideContext(Component, customContextTypes) {
+    // support decorator pattern
+    if (arguments.length === 0 || !arguments[0].displayName) {
```

Comment:
@mridgway another approach could be to just check if a function is passed in: `if (arguments.length === 0 || typeof arguments[0] !== 'function')`...